blockindex/bloom_range.go

```diff
@@ -11,24 +11,30 @@ type bloomRange struct {
 	bloom.BloomFilter
 }
 
-func newBloomRange(start uint64, bf bloom.BloomFilter) *bloomRange {
+func newBloomRange(bfSize, bfNumHash uint64) (*bloomRange, error) {
```

Comment:
good refactor, define it like this, then call `SetStart()` feels better/more clear